lib/resque/data_store.rb

```diff
@@ -91,10 +92,16 @@ def all_resque_keys
     end
 
     def server_time
-      time, _ = @redis.time
+      time, _ = @redis_time_available ? @redis.time : Time.now
       Time.at(time)
     end
 
+    def redis_time_available?
+      @redis.time
+    rescue Redis::CommandError
```

Comment:
The response is cached when the DataStore is initialized, so as long as you're using the same DataStore model it won't call this method again. I should probably also make this method private though.